lib/services/catalog/commands/index.js

```diff
@@ -14,13 +14,13 @@ module.exports = function (service) {
 		payload = _.cloneDeep(payload);
 
 		if (!payload.channel || !_.isString(payload.channel)) {
-			Promise.reject(new Error('payload.channel is required'));
+			throw new Error('payload.channel is required');
 		}
 		if (!payload.source || !_.isString(payload.source)) {
```

Comment:
Yup - Looks like this was addressed later in this PR by this commit: `commit b5a8158abaec86fd8891a74661932d4eafc25c5a`